src/main/java/io/github/debuggyteam/architecture_extensions/resource/ResourceUtils.java

```diff
@@ -0,0 +1,72 @@
+package io.github.debuggyteam.architecture_extensions.resource;
+
+import java.io.FileNotFoundException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.loader.api.ModContainer;
+
+import com.google.common.collect.Sets;
+
+import io.github.debuggyteam.architecture_extensions.ArchitectureExtensions;
+import io.github.debuggyteam.architecture_extensions.util.LRUHashMap;
+import net.minecraft.resource.ResourceType;
+
+public final class ResourceUtils {
+	// Field to get around weird java crap with lambdas
+	private static Exception exception;
```

Comment:
Word of warning; this won't play super well with multi-threaded scenarios if this is ever ran with multiple threads.

It'd be better to keep it in the method if possible, failing that, immediately execute the error handling or throw anyways.

Reply:
> Local variable exception defined in an enclosing scope must be final or effectively final
Can't put it in the method

Reply:
Then handle errors immediately if possible (log it), or leave as is and we'll try to deal with it another way later on. It's not a *critical* thing for vanilla, but could pose a problem if another mod decides to try to thread this.